commands/download.js

```diff
@@ -0,0 +1,76 @@
+var chalk = require('chalk')
+var prettyBytes = require('pretty-bytes')
+var Dat = require('../lib/dat')
+var logger = require('../lib/logger')
+var ui = require('../lib/ui')
+
+module.exports = function (args) {
+  var dat = Dat(args)
+  var log = logger(args)
+
+  var downloadTxt = 'Downloading '
+  var swarmTimeout = null
+
+  log.status('Starting Dat...\n', 0)
+  log.status('Connecting...', 1)
+
+  dat.on('error', onerror)
+
+  dat.on('ready', function () {
+    log.message('Downloading in ' + dat.dir + '\n')
+    dat.download(function (err) {
+      if (err) onerror(err)
```

Comment:
Yep! download-finished fires whenever the downloaded archive is complete ( i.e. can fire multiple times for live). This is the function that I call when things are finished:

```javascript
  function done () {
    self.emit('download-finished')
    if (!archive.live) cb(null)
  }
```